### PR TITLE
Highlight animation

### PR DESCRIPTION
**Pull request by Shushishtok**
- Adds a highlight animation that glows on and off while the UI is being highlighted. Note: because Valve can't do CSS properties right, the @keyframes property will throw errors at you if you have the CSS linter on. I could not find a way to fix that. Super annoying.
- Added an optional parameter to set the new highlight panel as the parent rather as a sibling. Optional, defaults to false.
- Changed `FindChild` to `FindChildTraverse` when crawling through the path, as sometimes providing the full path is not possible (due to classes and/or varying IDs, like when checking for the Tango's guide)
- Added highlighting on the Shop UI section for buying a Tango.

### CODE REVIEW

File: common/events.d.ts [hunk 1]
```diff
@@ -40,6 +40,7 @@ interface ClockTimeEvent {
 interface HighlightElementEvent {
     path: string;
     duration?: number;
+    setElementAsParent?: 0 | 1;
```
Comment (Perryvw) [inline common/events.d.ts:43]: This should be boolean

File: game/scripts/vscripts/util.ts [hunk 2]
```diff
@@ -151,9 +151,12 @@ export function displayDotaErrorMessage(message: string) {
  * Highlights a panel along a path
  * @param path The path along the ui to take, such as "HUDElements/lower_hud/center_with_stats/center_block/inventory"
  * @param duration Optional time in seconds after which to remove the highlight
+ * @param setElementAsParent Optional. Sets the element provided in the path as the parent, instead of as a sibling. Used for cases where the parent has the flow-children CSS property.
  */
-export function highlightUiElement(path: string, duration?: number) {
-    CustomGameEventManager.Send_ServerToAllClients("highlight_element", { path, duration });
+export function highlightUiElement(path: string, duration?: number, setElementAsParent?: boolean) {
+    let eventSetElementAsParent: 0 | 1
+    eventSetElementAsParent = setElementAsParent ? 1 : 0
```
Comment (Perryvw) [inline game/scripts/vscripts/util.ts:158]: remove this after changing the event field to boolean

Comment (Shushishtok) [reply game/scripts/vscripts/util.ts]: Done.

File: common/events.d.ts (hunk 1 above)
Comment (Shushishtok) [reply common/events.d.ts]: Changed to boolean.

**Pull request merged**